### PR TITLE
Created a rtk query for Challenges and replaced the useFetch implementation

### PR DESCRIPTION
**Pull request by sonai99**
This PR closes https://github.com/Real-Dev-Squad/website-status/issues/514
I have written the rtk query for GET_CHALLENGES_URL and have replaced the useFetch implementations
image
image

### CODE REVIEW

**Comment by sonai99** [comment]
@bhtibrewal please review this
@Pratiyushkumar

File: src/pages/challenges.tsx [hunk 1]
```diff
@@ -29,20 +27,16 @@ const renderCardList = (
...
     const [filteredChallenge, setFilteredChallenge] = useState<any>([]);
-    const { data: user } = useGetUserQuery(skipToken);
-    const { response, error, isLoading, callAPI } = useFetch(
-        CHALLENGES_URL,
-        {},
-        false
-    );
+    const { data, isLoading, error } = useGetChallengesQuery();
+    
+    const challengesResponse = data?.challenges;
```
Comment (Pratiyushkumar) [inline src/pages/challenges.tsx:32]: have you verfied the eif it is only error or isError key gettting destructured from useGetChallengesQuery()

File: src/interfaces/challenge.type.ts [hunk 2]
```diff
@@ -26,4 +26,4 @@ type challenge = {
     screen: string;
 };
 
-export default challenge;
+export default challenge;
```
Comment (Pratiyushkumar) [inline src/interfaces/challenge.type.ts:29]: can you explain this change please?

File: src/app/services/challengesApi.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { api } from './api';
+import challenge from '@/interfaces/challenge.type';
+
+export const challengesApi = api.injectEndpoints({
```
Comment (bhtibrewal) [inline src/app/services/challengesApi.ts:4]: Please write tests for this API.

File: src/pages/challenges.tsx [hunk 4]
```diff
@@ -29,20 +27,16 @@ const renderCardList = (
...
-        if (isLoggedIn && !Object.keys(response).length) {
-            callAPI();
-            if ('challenges' in response) {
-                const challenges: challenge['content'] = response.challenges;
+        if (isLoggedIn && !Object.keys(challengesResponse).length) {
+            if ('challenges' in challengesResponse) {
+                const challenges: challenge['content'] = data.challenges;
                 const challengeMap: any = [];
                 challengeMap.Active = challenges.filter(
                     (task) => task.is_active
```
Comment (bhtibrewal) [inline src/pages/challenges.tsx:42]: this filtering of task can be done in transformResponse attribute provided by RTK query

File: src/pages/challenges.tsx (hunk 1 above)
Comment (sonai99) [reply src/pages/challenges.tsx:32]: You are right, its isError, I have verified it. will fix,

**Comment by sonai99** [comment]
Raised a new PR - https://github.com/Real-Dev-Squad/website-status/pull/558
due to the change of the file structure

**Pull request closed without merging**